### ISSUE TITLE
[Combobox] VoiceOver listbox bugged (mobile)

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2859]
### Provide a general summary of the issue here

VoiceOver currently has a bug where the listbox behaves unexpectedly. When a combobox is opened either by selecting the combobox or by opening the list, the listbox cannot be focused. If focus is set manually to listbox, the button that opens the list can be focused, but the combobox cannot.

I don't currently have a Mac computer on hand, so I cannot verify whether this happens on desktop too.

Currently I have no specific information on why this happens. Needs more testing and/or discussion. 

### 🤔 Expected Behavior?

When selecting the combobox, the listbox opens and user is presented with a keyboard. User can move focus to the button that opens or closes the listbox. When listbox is opened, focus is set to the combobox textfield. User can then move to the listbox by moving forward first to the button and then to the listbox. User can the move forward or backward in the list, and select/deselect item(s).

If user sets focus to the listbox manually (e.g., tapping a listbox element using finger), focus is set to the listbox element and the user can move forward or backward in the list. If the user moves backwards to the combobox, the first element backwards from the list is the button that opens or closes the listbox, and then the combobox. 

### 😯 Current Behavior

When selecting the combobox, the listbox opens and user is presented with a keyboard. User can move focus to the button that opens or closes the listbox. When listbox is opened, focus is set to the combobox textfield. User cannot move to the listbox by moving forward or backward from the text field or the button that closes or opens the element. Focus move to element that is after the combobox instead.

If user sets focus to the listbox manually (e.g., tapping a listbox element using finger), focus is set to the listbox element and the user can move forward or backward in the list. If the user moves backwards out of the listbox, focus is set to element previous to the combobox and button. If user then moves back to the combobox, the listbox is focused instead.

If user moves forward and out of the listbox, the first element presented is the button that opens or closes the listbox. User cannot focus on the combobox at all. If user close the listbox, the combobox can be focused.

### 💁 Possible Solution

Currently I have no specific solution yet. 

My best guess at this point is, that the issue is related to how the element is constructed in HTML. VoiceOver tends to group parent and child elements together, and as both of the properties of the combobox and the listbox are inside the <sl-combobox> tag some of the elements might be grouped together incorrectly by VO. Rather than focusing on individual elements inside the <sl-combobox>, VoiceOver might be actually focusing on the <sl-combobox>, and thus miss some of the individual focusable items inside the element. 

VoiceOver tends to miss focusable and interactable elements when they are nested, and I believe this might be related.

### 🖥️ Steps to Reproduce

**Screen reader (VoiceOver mobile)**
Focus from combobox:
1. Set screen reader focus to the combobox 
2. Activate the combobox (-> listbox opens)
3. Move forward by swiping left, focus goes to the button that opens or closes the listbox
4. Move forward, listbox is ignored

Focus from listbox:
1. Tap on an item in listbox
2a. Move backwards from listbox, focus ignores button and combobox
2b. Move forwards from listbox, focus jumps to button, then to content after listbox

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome, Safari

### What operating system are you using?

iOS 26.0.1

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by a11ymiko** [comment]
This issue is probably caused by two or three factors combined. One of them is fact, that listbox is being rendered before '<input type='combobox'>' in DOM and that implies that if listbox was keyboard focusable it would be focused before combobox and not after it.

The second factor is `<slot part='wrapper'>` in which listbox is placed that has `tabindex='-1'`. Because of that listbox is only possible to receive focus when user uses arrow keys, which is not possible with iOS VoiceOver. 

The last factor is disappearing of listbox when user's focus leaves `<input role='combobox'>` component. So even when macOS VoiceOver user uses `Control + Option + arrow up/arrow down` to place reading focus on listbox focus is there only for split second but it quickly disappears as listbox disappears. This is shown on screen capture below.

Likely fix for that issue would be to make listbox not disappear when it receives VoiceOver reading focus when user uses `Control + Option + arrow up/arrow down`, remove `tabindex='-1'` from `<slot part='wrapper'>` and to move listbox in DOM to appear after `<input role="combobox">`, 

https://github.com/user-attachments/assets/142b94a8-e421-46c4-9b0f-f46465eb4398

**Comment by a11ymiko** [comment]
might be closed with: https://github.com/sl-design-system/components/issues/3256
